toml/src/main/kotlin/org/rust/toml/resolve/CargoTomlReferenceContributor.kt

```diff
@@ -30,7 +32,15 @@ class CargoTomlReferenceContributor : PsiReferenceContributor() {
             }
             registrar.registerReferenceProvider(onFeatureDependencyLiteral, CargoTomlFeatureDependencyReferenceProvider())
             registrar.registerReferenceProvider(onDependencyPackageFeature, CargoTomlDependencyFeaturesReferenceProvider())
-            registrar.registerReferenceProvider(dependencyGitUrl or packageUrl, CargoTomlUrlReferenceProvider())
+            // Starting from 2022.2 Toml plugin inserts web references in all string literals itself if needed
+            if (ApplicationInfo.getInstance().build < BUILD_222) {
+                registrar.registerReferenceProvider(dependencyGitUrl or packageUrl, CargoTomlUrlReferenceProvider())
+            }
```

Comment:
A similar functionality is implemented in the TOML plugin itself for all plain text url in literals to have the consistent behaviour between all config languages like properties, json, yaml
See https://youtrack.jetbrains.com/issue/IDEA-293807/Make-plain-text-URLs-as-clickable-in-TOML-string-value

I didn't remove the corresponding tests to be sure that important cases work as expected for user even without reference provider from Rust plugin side

CC @Kobzol